src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py

```diff
@@ -0,0 +1,345 @@
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        starting_frames: Optional[list[int]] = None,
+        parent_container: Literal["acquisition", "processing/behavior"] = "acquisition",
+        module_description: Optional[str] = None,
+    ):
+        """
+        Convert the video data file(s) to :py:class:`~pynwb.image.ImageSeries` and write them in the
+        :py:class:`~pynwb.file.NWBFile`. Data is written in a single :py:class:`~pynwb.image.ImageSeries` container with
+        a path to each external file.
+
+        Parameters
+        ----------
+        nwbfile : NWBFile, optional
+            nwb file to which the recording information is to be added
+        metadata : dict, optional
+            Dictionary of metadata information such as name and description of the video. The key must correspond to
+            the video_name specified in the constructor.
+            Should be organized as follows::
+
+                metadata = dict(
+                    Behavior=dict(
+                        Video=dict(
+                            ExternalVideo=dict(
+                                description="Description of the video..",
+                                unit="Frames",
+                                ...,
+                            ),
+                        )
+                    )
+                )
+
+            and may contain most keywords normally accepted by an ImageSeries
+            (https://pynwb.readthedocs.io/en/stable/pynwb.image.html#pynwb.image.ImageSeries).
+        starting_frames : list, optional
+            List of start frames for each video written using external mode.
+            Required if more than one path is specified.
+        parent_container: {'acquisition', 'processing/behavior'}
+            The container where the ImageSeries is added, default is nwbfile.acquisition.
+            When 'processing/behavior' is chosen, the ImageSeries is added to nwbfile.processing['behavior'].
+        module_description: str, optional
+            If parent_container is 'processing/behavior', and the module does not exist,
+            it will be created with this description. The default description is the same as used by the
+            conversion_tools.get_module function.
+        """
+        metadata = metadata or dict()
+
+        file_paths = self.source_data["file_paths"]
+
+        # Be sure to copy metadata at this step to avoid mutating in-place
+        videos_metadata = deepcopy(metadata).get("Behavior", dict()).get("Video", None)
+        if videos_metadata is None:
+            videos_metadata = deepcopy(self.get_metadata()["Behavior"]["Video"])
+        image_series_kwargs = metadata["Behavior"]["Video"][self.video_name]
+        image_series_kwargs["name"] = self.video_name
+
+        timing_type = self.get_timing_type()
```

Comment:
Probably not for this PR, but we should unify this behavior with the other interfaces. The default should be to check whether the timestamps are uniform and, if so, write starting_time and rate. This behavior can then be overridden with the always_write_timestamps option.

References for current implementations:

Ecephys:
https://github.com/catalystneuro/neuroconv/blob/2f9661ff4c3920b39c5eef2bce8063751390784f/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py#L323-L335

Ophys:
https://github.com/catalystneuro/neuroconv/blob/2f9661ff4c3920b39c5eef2bce8063751390784f/src/neuroconv/datainterfaces/ophys/baseimagingextractorinterface.py#L140-L150

Reply:
Yeah, makes sense to me, but I think beyond the scope of this PR.

Reply:
Yes,I was just checking with you if there would not be major roadblocks as you are reading this on detail.